person.rb

```diff
@@ -27,4 +31,9 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
- [ Required ] I would like to suggest that you refactor this method to take a `book` , `date` as arguments and then utilize the `Rental` class constructor to create a rental, this will ensure that `Book` is set in `Rental` constructor. You can do this by refactoring your code as shown below.

```ruby
   def add_rental(book, date)
      Rental.new(date, book, self)
   end
```